### PR TITLE
sysutils/FTP backup plugin

### PR DESCRIPTION
**Pull request by manhofmann**
Adds support to backup config to FTP server

### CODE REVIEW

**Comment by manhofmann** [comment]
Hello @fraenki 
any chance to look into this pull request?
Best regards
Manuel

**Comment by fraenki** [comment]
@manhofmann sorry, not my area of expertise :)

File: sysutils/ftp-backup/src/opnsense/mvc/app/library/OPNsense/Backup/FTP.php [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        fseek($infile, 0); 
+    
+        curl_setopt_array($curl, array(
+            CURLOPT_URL => $url . $filename,
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_TIMEOUT => 60,
+            CURLOPT_UPLOAD => 1,
+            CURLOPT_INFILE => $infile,
+            CURLOPT_INFILESIZE => strlen($file_content),
+            CURLOPT_USERPWD => $username . ":" . $password,
```
Comment (fabianfrz) [inline sysutils/ftp-backup/src/opnsense/mvc/app/library/OPNsense/Backup/FTP.php:218]: There seems to be another interesting option worth to try:
CURLOPT_FTP_CREATE_MISSING_DIRS

File: sysutils/ftp-backup/pkg-descr [hunk 2]
```diff
@@ -0,0 +1,4 @@
+This package adds a backup option using a FTP server.
+
+Due to the sensitive nature of the data being send to the backup, we
+strongly advise to not use a public service to send backups to.
```
Comment (fabianfrz) [inline sysutils/ftp-backup/pkg-descr:4]: I would also mention the optional usage of TLS (at least on the data connection).

**Review by fabianfrz — CHANGES_REQUESTED** [review]
I added some comments of some things I have seen.

Additionally, FTP servers in contrast to Nextcloud, do not necessary have an automatically handled file retention (if configured). This means that you may implement a file rotation system like in the Google Drive backup if you like.

**Comment by AdSchellevis** [comment]
no feedback

**Pull request closed without merging**